tests/unit_tests/crypto.cpp

```diff
@@ -61,6 +69,26 @@ namespace
     out << "BEGIN" << value << "END";  
     return out.str() == "BEGIN<" + std::string{expected, sizeof(T) * 2} + ">END";
   }
+
+  bool keccak_harness()
+  {
+    size_t inlen = sizeof(source);
+    int mdlen = (int)sizeof(md);
+    int ret = keccak(source, inlen, md, mdlen);
+
+    if (md[0] != 0x00)
+    {
+        return true;
+    }
+    else if (!ret)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+  }
```

Comment:
That doesn't seem to be used ? In any case, there are crypto tests in tests/crypto already.

Reply:
Didn't see any tests for the keccak library, should this go in its own folder/test file?

Reply:
If there is none, then a keccak test could go in crypto tests, sure.